collect_app/src/main/java/org/odk/collect/android/activities/BearingActivity.java

```diff
@@ -58,55 +55,43 @@ protected void onPause() {
 
         sensorManager.unregisterListener(this, accelerometer);
         sensorManager.unregisterListener(this, magnetometer);
-
-        if (bearingDialog != null && bearingDialog.isShowing()) {
-            bearingDialog.dismiss();
-        }
+        DialogFragmentUtils.dismissDialog(MaterialProgressDialogFragment.class, getSupportFragmentManager());
     }
 
     @Override
     protected void onResume() {
         super.onResume();
         sensorManager.registerListener(this, accelerometer, SensorManager.SENSOR_DELAY_GAME);
         sensorManager.registerListener(this, magnetometer, SensorManager.SENSOR_DELAY_GAME);
-        bearingDialog.show();
+        setupBearingDialog();
     }
 
     /**
      * Sets up the look and actions for the progress dialog while the compass is
      * searching.
      */
     private void setupBearingDialog() {
-        // dialog displayed while fetching bearing
-        bearingDialog = new DayNightProgressDialog(this);
         DialogInterface.OnClickListener geopointButtonListener =
-                new DialogInterface.OnClickListener() {
-                    @Override
-                    public void onClick(DialogInterface dialog, int which) {
-                        switch (which) {
-                            case DialogInterface.BUTTON_POSITIVE:
-                                returnBearing();
-                                break;
-                            case DialogInterface.BUTTON_NEGATIVE:
-                                bearingDecimal = null;
-                                finish();
-                                break;
-                        }
+                (dialog, which) -> {
+                    switch (which) {
+                        case DialogInterface.BUTTON_POSITIVE:
+                            returnBearing();
+                            break;
+                        case DialogInterface.BUTTON_NEGATIVE:
+                            bearingDecimal = null;
+                            finish();
+                            break;
                     }
                 };
 
-        // back button doesn't cancel
-        bearingDialog.setCancelable(false);
-        bearingDialog.setIndeterminate(true);
-        bearingDialog.setIcon(android.R.drawable.ic_dialog_info);
-        bearingDialog.setTitle(getString(R.string.getting_bearing));
-        bearingDialog.setMessage(getString(R.string.please_wait_long));
-        bearingDialog.setButton(DialogInterface.BUTTON_POSITIVE,
-                getString(R.string.accept_bearing),
-                geopointButtonListener);
-        bearingDialog.setButton(DialogInterface.BUTTON_NEGATIVE,
-                getString(R.string.cancel_location),
-                geopointButtonListener);
+        MaterialProgressDialogFragment dialog = new MaterialProgressDialogFragment();
+        dialog.setCancelable(false);
+        dialog.setIcon(android.R.drawable.ic_dialog_info);
```

Comment:
It would be nice if methods in MaterialProgressDialogFragment returned the dialog itself the instead of this we could have chaining like: 
```
dialog.setCancelable(false);
   .setIcon(android.R.drawable.ic_dialog_info);
   .setTitle(getString(R.string.getting_bearing));
   .setMessage(getString(R.string.please_wait_long));
   .setPositiveButton(getString(R.string.accept_bearing), geopointButtonListener);
   .setNegativeButton(getString(R.string.cancel_location), geopointButtonListener);
```

Reply:
Maybe we should introduce a builder as it seems like we're moving towards that kind of pattern. It could look like:

```
new MaterialProgressDialogFragment.Builder()
   .setCancelable(false)
   .setIcon(android.R.drawable.ic_dialog_info)
   .setTitle(getString(R.string.getting_bearing))
   .setMessage(getString(R.string.please_wait_long))
   .setPositiveButton(getString(R.string.accept_bearing), geopointButtonListener)
   .setNegativeButton(getString(R.string.cancel_location), geopointButtonListener)
   .build();

Reply:
@seadowg 
Something like [MaterialProgressDialogFragmentNew](https://github.com/getodk/collect/pull/4993/files#diff-1c77f14689d2d89daa2ff00faab89f24800f3253c3dd47745bc6da5fd82f4f03) it's a solution that allows us to use the builder patter and also create dialogs using its constructor because we need this in dialogs that extend `MaterialProgressDialogFragmentNew`.
I Used that new `MaterialProgressDialogFragmentNew` in a few places to show how it looks like. What do you think?

Reply:
That won't work with rotatations/recreations - remember Fragments will be recreated using their no arg constructors so like Activity objects, storing state in fields/overriding constructors will cause problems. That's the reason for the use of args in `MaterialProgressDialogFragment`. You'll see that there are tests to make sure the state is carried between recreated instances in `MaterialProgressDialogFragmentTest`.

Reply:
Right I knew I was missing something... actually I was a bit tricked because when I asked to create separate classes for those dialogs that previously used `DayNightProgressDialog` I was right. Most of those dialogs have listener that should be used with `POSITIVE`/`NEGATIVE` buttons but we can't support such listeners without subclassing `MaterialProgressDialogFragment` because it won't work with recreation.
In this case this whole pr contains a lot of bugs like that and I think we should just close it for now. I can get back to it later when I'm free.
So if we need to support listeners we need sub classes for all the other cases we can add a simple builder.